Added docs and guide for implementation http-tools in Magento 2 (#5)
phpro/http-tools: docs/framework/magento2.md

# Using this package inside Magento2

## Configuration via di.xml
1. Define and configure your custom plugins/middlewares or implement some of [HTTPlug](http://docs.php-http.org/en/latest/plugins/)
2. Use `Phpro\HttpTools\Client\Factory\LazyClientLoader` to configure your preferred HTTP client (`AutoDiscoveredClientFactory`, `GuzzleClientFactory`, `SymfonyClientFactory`, ...) and define options such as base_uri, default headers, ...

> 3. Always create your own _Transport_ class (as kind of decorator) because Magento don't know the concept of factories or Symfony stack). Inside your custom _Transport_ you can use the built-in `JsonTransport`

Reviewer: ```suggestion
3. Always create your own _Transport_ class as decorator. Magento don't know the concept of factories or stack in its DI component. Inside your custom _Transport_ you can use the built-in `JsonTransport`
```

        return WeatherResponse::fromRawArray(
            ($this->transport)($weatherRequest)
        );
    }
}
```

## Example Request Value Object
```php
<?php
declare(strict_types=1);

namespace Mage\OpenWeather\Http\Request;

use Phpro\HttpTools\Request\RequestInterface;

class WeatherRequest implements RequestInterface
{

    public function method(): string
    {
        return 'GET';
    }

    public function uri(): string
    {
        return 'data/2.5/weather?q={location}&APPID={appid}';
    }

    public function uriParameters(): array
    {
        return [
            'location' => 'Antwerp,be',
            'appid' => '...'
        ];
    }

    public function body(): array
    {
        return [];
    }
}
```

## Example Response Value Object
```php
<?php
declare(strict_types=1);

namespace Mage\OpenWeather\Http\Response;

class WeatherResponse
{
    private $type;

    private $description;

    public function __construct(string $type, string $description)
    {
        $this->type = trim($type);
        $this->description = trim($description);
    }

    public static function fromRawArray(array $data): self
    {

>         if (!isset($data['weather']) || !is_array($data['weather']) || !isset($data['weather'][0]['main'])) {

Reviewer: IMO it's better to normalize the data and cast it into the correct type instead of throwing a real excetion here.
Unless it is a super blocking api call.

E.g. if the weather array is not available, you could fall back to 2 empty strings.
On the other hand: if weather is not available, maybe there is an "error" property you can as  better error message?
